File: LeetCode/Easy/0119-pascals-triangle-ii/0119-pascals-triangle-ii.cs
public class Solution {
    private bool doIterative = false;
    public IList<int> GetRow(int rowIndex) {
        if (doIterative)
        {
            var row = new int[rowIndex + 1];
            for (var i = 0; i <= rowIndex; i++) 
            {
                row[0] = 1;
                row[i] = 1;
                for (var j = i - 1; j >= 1; j--)
                {                
                    row[j] = row[j - 1] + row[j];
                }
            }
            
            return row;
        }
        else
        {
            var row = new int[rowIndex + 1];
            row[0] = 1;
            for (var i = 1; i < rowIndex + 1; i++)
            {
                row[i] = (int)(1l * row[i - 1] * (rowIndex - i + 1) / i);
            }
            return row;
        }
    }
}

File: LeetCode/Easy/0119-pascals-triangle-ii/0119-pascals-triangle-ii.py

class Solution:
    # # classic, DP solution
    # def getRow(self, rowIndex: int) -> List[int]:
    #     row = [1] * (rowIndex + 1)

    #     for r in range(2, rowIndex + 1):
    #         for c in range(r - 1, 0, -1):
    #             row[c] += row[c - 1]
            
    #     return row

    # binomial coefficient solution
    def getRow(self, rowIndex: int) -> List[int]:
        row = [1] * (rowIndex + 1)
        
        for i in range(1, rowIndex + 1):
            # Multiply first, then integer division to prevent truncation
            row[i] = row[i - 1] * (rowIndex - i + 1) // i
            
        return row